include/libtorrent/sha512.hpp

```diff
@@ -0,0 +1,23 @@
+#ifndef TORRENT_SHA512_HPP_INCLUDED
+#define TORRENT_SHA512_HPP_INCLUDED
+
+#include "libtorrent/config.hpp"
+#include <cstdint>
+
+namespace libtorrent
+{
+
+	struct sha512_ctx
+	{
+		std::uint64_t length, state[8];
+		size_t curlen;
+		std::uint8_t buf[128];
+	};
+
+	TORRENT_EXTRA_EXPORT int SHA512_init(sha512_ctx* context);
```

Comment:
The original code, both from `sha1` and `sha512` already provide this interface, it was just slightly accommodated to mimic libcrypto, do you have any particular idea here?